blackd.py

```diff
@@ -51,22 +52,42 @@ def make_app() -> web.Application:
             )
         except ValueError:
             return web.Response(status=400, text="Invalid line length header value")
-        py36 = False
         pyi = False
+        versions: Set[black.TargetVersion] = set()
         if PYTHON_VARIANT_HEADER in request.headers:
             value = request.headers[PYTHON_VARIANT_HEADER]
             if value == "pyi":
                 pyi = True
             else:
-                try:
-                    major, *rest = value.split(".")
-                    if int(major) == 3 and len(rest) > 0:
-                        if int(rest[0]) >= 6:
-                            py36 = True
-                except ValueError:
-                    return web.Response(
-                        status=400, text=f"Invalid value for {PYTHON_VARIANT_HEADER}"
-                    )
+                for version in value.split(","):
+                    tag = "cpy"
+                    if version.startswith("cpy"):
+                        version = version[len("cpy") :]
+                    elif version.startswith("pypy"):
+                        tag = "pypy"
+                        version = version[len("pypy") :]
+                    major_str, *rest = version.split(".")
+                    try:
+                        major = int(major_str)
+                        if len(rest) > 0:
+                            minor = int(rest[0])
+                        else:
+                            # Default to lowest supported minor version.
+                            minor = 7 if major == 2 else 3
+                        version_str = f"{tag.upper()}{major}{minor}"
+                        # If PyPY is the same as CPython in some version, use
+                        # the corresponding CPython version.
+                        if tag == "pypy" and not hasattr(
+                            black.TargetVersion, version_str
+                        ):
+                            version_str = f"CPY{major}{minor}"
+                        versions.add(black.TargetVersion[version_str])
+                    except (KeyError, ValueError):
+                        return web.Response(
+                            status=400,
+                            text=f"Invalid value for {PYTHON_VARIANT_HEADER}",
```

Comment:
Let's have a more specific error message here. End users will likely encounter this if they misconfigure their clients

Reply:
Will do